dragon/console_prompt.rb

```diff
@@ -63,11 +67,45 @@ def move_cursor_left
         update_cursor_position_px
       end
 
+      # ? Works well enough, could be better
+      # TODO: Maybe possibly add skipping multiple word-breaking characters
+      # TODO: (requires re-writing a lot of the code most likely)
```

Comment:
Maybe not so much....

I think something like this could work for example for `move_cursor_left_word`

```rb
def move_cursor_left_word
  return if @cursor_position.zero?
  
  new_position = @current_position - 1
  previous_character_is_boundary = is_word_boundary? @current_input_str[@current_position - 1]
  if previous_character_is_boundary
    new_position -=1 until !is_word_boundary?(@current_input_str[new_position]) || new_position.zero?
  else
    new_position -=1 until is_word_boundary?(@current_input_str[new_position]) || new_position.zero?
  end
  
  @cursor_position = new_position
  update_cursor_position_px
end
```

And similar for move_word_right

Reply:
I've slept and will now try to make sense of this :>
I think I need to write the `is_word_boundary?` function myself don't I?

Reply:
Ok `is_word_boundary?` has been implemented, needed to tweak your code a little (beside shortening it) 
```rb
previous_character_is_boundary = is_word_boundary? @current_input_str[@current_position - 1]
```
became 
```rb
is_word_boundary? @current_input_str[new_pos]
``` 
which is instantly used in a ternary. 
```rb
new_position -=1 until !is_word_boundary?(@current_input_str[new_position]) || new_position.zero?
``` 
this had to be changed to this 
```rb
new_pos -= 1 until !(is_word_boundary? @current_input_str[new_pos - 1]) || new_pos.zero?
``` 
because it always overshot one character, but thank you, I wouldn't figured it out that quickly without your help

Reply:
Now to insert that idea into `move_cursor_right_word`